tests/phpunit/tests/post/wpUpdatePost.php

```diff
@@ -0,0 +1,260 @@
+
+		wp_update_post( $post );
+
+		$tags = wp_get_post_tags( $post->ID, array( 'fields' => 'ids' ) );
+		$this->assertSameSets( array( $tag_1['term_id'], $tag_2['term_id'] ), $tags );
+
+		wp_update_post(
+			array(
+				'ID'         => $post->ID,
+				'tags_input' => array( $tag_2['term_id'], $tag_3['term_id'] ),
+			)
+		);
+
+		$tags = wp_get_post_tags( $post->ID, array( 'fields' => 'ids' ) );
+		$this->assertSameSets( array( $tag_2['term_id'], $tag_3['term_id'] ), $tags );
+	}
+
+	/**
+	 * Test the wp_update_post() filters post content when 'post_status' is 'draft'.
+	 *
+	 * @ticket 22944
+	 */
+	public function test_wp_update_post_with_content_filtering() {
+		kses_remove_filters();
+
+		$post_id = wp_insert_post(
+			array(
+				'post_title' => '<script>Test</script>',
+			)
+		);
+		$post    = get_post( $post_id );
+		$this->assertSame( '<script>Test</script>', $post->post_title );
+		$this->assertSame( 'draft', $post->post_status );
+
+		kses_init_filters();
+
+		wp_update_post(
+			array(
+				'ID'          => $post->ID,
+				'post_status' => 'publish',
+			)
+		);
+
+		kses_remove_filters();
+
+		$post = get_post( $post->ID );
+		$this->assertSame( 'Test', $post->post_title );
+	}
+
+	/**
+	 * Test updating a post and preserving post_date when changing to future status.
+	 *
+	 * @ticket 62468
+	 *
+	 * @dataProvider data_update_post_preserves_date_for_future_posts
+	 *
+	 * @param string $initial_status  Initial post status.
+	 * @param string $time_offset     Time offset.
+	 * @param string $expected_status Expected post status.
+	 */
+	public function test_update_post_preserves_date_for_future_posts( $initial_status, $time_offset, $expected_status ) {
+
+		$post_id = self::factory()->post->create(
+			array(
+				'post_status' => $initial_status,
+			)
+		);
+
+		$future_date = gmdate( 'Y-m-d H:i:s', strtotime( $time_offset ) );
+		$update_data = array(
+			'ID'          => $post_id,
+			'post_status' => 'future',
+			'post_date'   => $future_date,
+		);
+
+		wp_update_post( $update_data );
+		$updated_post = get_post( $post_id );
+
+		$this->assertSame( $future_date, $updated_post->post_date );
+		$this->assertSame( $expected_status, $updated_post->post_status );
+	}
+
+	/**
+	 * Data provider for test_update_post_preserves_date_for_future_posts.
+	 *
+	 * @return array[] Test parameters.
+	 */
+	public function data_update_post_preserves_date_for_future_posts() {
+		return array(
+			'pending to future with 1 day more' => array(
+				'initial_status'  => 'pending',
+				'time_offset'     => '+1 day',
+				'expected_status' => 'future',
+			),
+			'draft to future with 1 day more'   => array(
+				'initial_status'  => 'draft',
+				'time_offset'     => '+1 day',
+				'expected_status' => 'future',
+			),
+			'publish to future with 1 day more' => array(
+				'initial_status'  => 'publish',
+				'time_offset'     => '+1 day',
+				'expected_status' => 'publish',
```

Comment:
I don't think this is correct.

On trunk:

1. Create and publish a post
2. Change the time to tomorrow
3. Change the status to scheduled
4. Click save
5. Observe in the database that the new status is `future`